Source/swiftlint/Commands/LintCommand.swift

```diff
@@ -55,7 +55,10 @@ struct LintCommand: CommandType {
             violations += currentViolations
             reporter.reportViolations(currentViolations, realtimeCondition: true)
         }.flatMap { files in
-            reporter.reportViolations(violations, realtimeCondition: false)
+            if isWarningThresholdBroken(configuration, violations: violations) {
+                violations.append(createThresholdViolation(configuration.warningThreshold!))
+            }
+            reporter.reportViolations(violations, realtimeCondition: true)
```

Comment:
Yup so changed to false and the error no longer shows up. It still adds to the array but then I get 

"Command /bin/sh failed with exit code 2" whereas if it is true I get an error shown and then the done linting comment (in xcode)